thundra/wrappers/flask/flask_wrapper.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import traceback
+from functools import wraps
+
+from thundra import constants
+from thundra.application.global_application_info_provider import GlobalApplicationInfoProvider
+from thundra.config import config_names
+from thundra.config.config_provider import ConfigProvider
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.context.plugin_context import PluginContext
+from thundra.context.tracing_execution_context_provider import TracingExecutionContextProvider
+from thundra.wrappers import wrapper_utils, web_wrapper_utils
+from thundra.wrappers.base_wrapper import BaseWrapper
+from thundra.wrappers.flask import flask_executor
+
+logger = logging.getLogger(__name__)
+
+
+class FlaskWrapper(BaseWrapper):
+
+    def __init__(self, api_key=None, disable_trace=False, disable_metric=True, disable_log=True, opts=None):
+        super(FlaskWrapper, self).__init__(api_key, disable_trace, disable_metric, disable_log, opts)
+        self.application_info_provider = GlobalApplicationInfoProvider()
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider())
+        self.plugin_context = PluginContext(application_info=self.application_info_provider.get_application_info(),
+                                            request_count=0,
+                                            executor=flask_executor,
+                                            api_key=self.api_key)
+
+        self.plugins = wrapper_utils.initialize_plugins(self.plugin_context, disable_trace, disable_metric, disable_log,
+                                                        config=self.config)
+
+        web_wrapper_utils.update_application_info(self.application_info_provider, self.plugin_context.application_info,
+                                                  constants.ClassNames['FLASK'])
+
+    def before_request(self, request):
+        # Execution context initialization
+        execution_context = wrapper_utils.create_execution_context()
+        execution_context.platform_data['request'] = request
+
+        # Execute plugin hooks before running user's handler
+        self.plugin_context.request_count += 1
+        self.execute_hook('before:invocation', execution_context)
+        return execution_context
+
+    def after_request(self, response):
+        try:
+            execution_context = ExecutionContextManager.get()
```

Comment:
I think we should stora execution context in Flask's `ApplicationContext` at `before_request` and get execution context from there here. So, even if there is a problem while propagating execution context by `ExecutionContextManager`, we will still be able to get the correct one always.

https://flask.palletsprojects.com/en/1.1.x/appcontext/